src/containers/BooksForm.js

```diff
@@ -19,21 +34,75 @@ const chooseACategory = () => {
   );
 };
 
-const BooksForm = () => (
-  <form action="">
-    <h2>Insert new book</h2>
-    <input
-      placeholder="Enter Book Title"
-      type="text"
-      name="title"
-    />
+class BooksForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      book: {
+        title: '',
+        category: categories[0],
+      },
+    };
 
-    <chooseACategory />
-    <button type="button" id="add-a-book">
-      Add Book
-    </button>
-  </form>
-);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleBookSubmit = this.handleBookSubmit.bind(this);
+  }
 
+  handleChange({ target: { value, name } }) {
+    const { book } = this.state;
+    this.setState({
+      book: {
+        ...book,
+        [name]: value,
+      },
+    });
```

Comment:
hello there is really no need for getting the state, the setState is meant to change it anyway.
```
this.setState({
      book: {
        [name]: value,
      },
    });
```

is  Just fine.